### PR TITLE
DSD-642: Removing the showLabel prop from SearchBar

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-642](https://jira.nypl.org/browse/DSD-642)

## This PR does the following:
- Removes the `showLabel` prop from the `SearchBar` component. This prop was never used in the component so it's confusing seeing that prop and it not visually affecting the component. Internally, the labels for the `Select` and `TextInput` components are never shown, so the high-level `showLabel` is not necessary.

## How has this been tested?

Storybook and tests.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr796-80qtb7wve8n336mjpe1chcir2qrwngmd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61af712a6e958afa4ef028f5

File: src/components/SearchBar/SearchBar.stories.mdx [hunk 1]
```diff
@@ -213,9 +216,9 @@ precedence.
   <Story
     name="Search Autocomplete"
     args={{
-      isInvalid: false,
       isDisabled: false,
       isRequired: false,
+      isInvalid: false,
```
Comment (bigfishdesign13) [inline src/components/SearchBar/SearchBar.stories.mdx:220]: Nit: `isInvalid` should be between `isDisabled` and `isRequired`.

**Review by bigfishdesign13 — APPROVED** [review]
Approved, with one minor comment.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr796-80qtb7wve8n336mjpe1chcir2qrwngmd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61af712a6e958afa4ef028f5

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr796-80qtb7wve8n336mjpe1chcir2qrwngmd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61af712a6e958afa4ef028f5

**Pull request merged**